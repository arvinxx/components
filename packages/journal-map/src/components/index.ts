export { default as Chart } from './Chart';
export { default as Flow } from './Flow';
