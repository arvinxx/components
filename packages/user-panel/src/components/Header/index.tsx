import React from 'react';
import type { FC } from 'react';
import { Typography } from 'antd';

import { useFormatMessage } from '../../hooks';
import type { LocaleKey } from '../../type';

import './style.less';

const { Title } = Typography;

export interface HeaderProps {
  type: 'login' | 'register' | 'forgot';
  logo?: string;
  url?: string;
}

const Header: FC<HeaderProps> = ({ type, logo, url }) => {
  const f = useFormatMessage();

  return (
    <div className="avx-user-panel-header">
      {!logo ? null : (
        <div className="avx-user-panel-logo">
          <a href={url}>
            <img alt={'logo'} src={logo} />
          </a>
        </div>
      )}

      <Title
        level={2}
        style={{
          textAlign: 'center',
          marginBottom: 24,
          fontWeight: 'normal',
        }}
      >
        {f(`${type}.header` as LocaleKey)}
      </Title>
    </div>
  );
};

export default Header;
