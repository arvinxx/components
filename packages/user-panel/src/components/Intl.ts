import createI18n from '@arvinxu/i18n';
import localeMessageMaps from '../locales';

export const { IntlProvider, useFormatMessage } = createI18n(localeMessageMaps);
