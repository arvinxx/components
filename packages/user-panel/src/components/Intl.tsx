import React from 'react';
import { IntlProvider as IntlProvider_ } from 'react-intl';
import type { LocaleMessages } from '../type';

export const IntlProvider: React.FC<
  Omit<React.ComponentProps<typeof IntlProvider_>, 'messages'> & {
    messages: LocaleMessages;
  }
> = (props) => <IntlProvider_ {...props} />;
