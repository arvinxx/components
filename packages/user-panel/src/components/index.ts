export { default as IntlWrapper } from './IntlWrapper';
export { default as Header, HeaderProps } from './Header';
export * from './Intl';
