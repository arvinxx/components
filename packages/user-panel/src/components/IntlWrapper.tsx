import React from 'react';
import type { FC } from 'react';
import { useI18n } from '../hooks';
import { IntlProvider } from './Intl';

const IntlWrapper: FC = ({ children }) => {
  const { locale, messages } = useI18n();
  return (
    <IntlProvider defaultLocale={'zh-CN'} locale={locale} messages={messages}>
      {children}
    </IntlProvider>
  );
};
export default IntlWrapper;
