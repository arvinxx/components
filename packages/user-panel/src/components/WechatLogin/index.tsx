import type { FC } from 'react';
import React from 'react';
import { WechatOutlined } from '@ant-design/icons';
import { Button, message } from 'antd';

import { useFormatMessage } from '../Intl';

import './style.less';

interface IWechatLoginProps {
  disabled?: boolean;
  login?: () => void;
}
const WechatLogin: FC<IWechatLoginProps> = ({ login }) => {
  const f = useFormatMessage();
  const handleWeChatClick = () => {
    if (login) {
      login();
    } else {
      const errorMsg = f('login.wechat-login.function.error');

      message.error(errorMsg);
    }
  };

  return (
    <Button
      size="large"
      onClick={handleWeChatClick}
      className="avx-user-panel-wechat"
      type="default"
      data-testid={'wechat-btn'}
    >
      <WechatOutlined className="avx-user-panel-wechat-icon" />
      {f('login.wechat-login.btn')}
    </Button>
  );
};

export default WechatLogin;
