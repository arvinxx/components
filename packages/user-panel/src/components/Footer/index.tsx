import React from 'react';
import type { FC } from 'react';
import type * as H from 'history';
import { useHistory } from 'react-router';
import { Typography, Divider } from 'antd';

import WechatLogin from '../WechatLogin';

import { useFormatMessage } from '../Intl';
import type { PanelContentType } from '../../types';

import './style.less';

const { Text } = Typography;

export interface FooterProps {
  /**
   * 类型
   */
  type: PanelContentType;
  /**
   * @title 微信登录方法
   */
  onWechatLoginClick?: () => void;

  /**
   * @title 注册页面 Url
   */
  onRegisterClick?: (history: H.History) => void;
}

export const Footer: FC<FooterProps> = ({
  type,
  onWechatLoginClick,
  onRegisterClick,
}) => {
  const f = useFormatMessage();

  const history = useHistory();

  const onClick = () => {
    switch (type) {
      case 'login':
        onRegisterClick?.(history);
        break;
      case 'register':
        break;
      case 'forgot':
        break;
      default:
        return '';
    }
  };
  return (
    <div className="avx-user-panel-footer">
      <Divider dashed className="avx-user-panel-footer-divider">
        {f('login.or')}
      </Divider>
      {onWechatLoginClick ? <WechatLogin login={onWechatLoginClick} /> : null}
      <div className="avx-user-panel-footer-other">
        <Text>
          {f(type === 'login' ? 'login.no-account' : 'register.have-account')}
        </Text>
        <div className="avx-user-panel-footer-register">
          <a onClick={() => onClick()}>
            {f(type === 'login' ? 'login.signup' : 'register.sign-in')}
          </a>
        </div>
      </div>
    </div>
  );
};
