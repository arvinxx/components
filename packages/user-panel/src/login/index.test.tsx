import '../../tests/matchMedia.mock';

import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';

import Login from './index';
import { IntlProvider } from '../components';

describe('Login 组件', () => {
  it('默认状态', () => {
    const { container } = render(
      <IntlProvider>
        <Login />
      </IntlProvider>,
    );
    expect(container).toMatchSnapshot();
  });

  it('获取验证码', async () => {
    const { container, findByText } = render(
      <IntlProvider>
        <Login />
      </IntlProvider>,
    );
    const btn = await findByText('手机号登录');

    await act(async () => {
      fireEvent.click(btn);
      const captcha = await findByText('获取验证码');
      await fireEvent.click(captcha);
    });

    expect(container).toMatchSnapshot();
  });
});
