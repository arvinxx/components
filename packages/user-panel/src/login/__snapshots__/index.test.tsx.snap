// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login 组件 获取验证码 1`] = `
<div>
  <div
    class="avx-user-panel-login-container"
  >
    <form
      autocomplete="off"
      class="ant-form ant-form-vertical"
    >
      <input
        style="display: none;"
        type="text"
      />
      <div
        class="ant-tabs ant-tabs-top ant-tabs-card"
      >
        <div
          class="ant-tabs-nav"
          role="tablist"
        >
          <div
            class="ant-tabs-nav-wrap"
          >
            <div
              class="ant-tabs-nav-list"
              style="transform: translate(0px, 0px);"
            >
              <div
                class="ant-tabs-tab"
              >
                <div
                  aria-controls="rc-tabs-test-panel-account"
                  aria-selected="false"
                  class="ant-tabs-tab-btn"
                  id="rc-tabs-test-tab-account"
                  role="tab"
                  tabindex="0"
                >
                  账户密码登录
                </div>
              </div>
              <div
                class="ant-tabs-tab ant-tabs-tab-active"
              >
                <div
                  aria-controls="rc-tabs-test-panel-mobile"
                  aria-selected="true"
                  class="ant-tabs-tab-btn"
                  id="rc-tabs-test-tab-mobile"
                  role="tab"
                  tabindex="0"
                >
                  手机号登录
                </div>
              </div>
              <div
                class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
          <div
            class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
          >
            <button
              aria-controls="rc-tabs-test-more-popup"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-hidden="true"
              class="ant-tabs-nav-more"
              id="rc-tabs-test-more"
              style="visibility: hidden; order: 1;"
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="ellipsis"
                class="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="ant-tabs-content-holder"
        >
          <div
            class="ant-tabs-content ant-tabs-content-top"
          >
            <div
              aria-hidden="true"
              aria-labelledby="rc-tabs-test-tab-account"
              class="ant-tabs-tabpane"
              id="rc-tabs-test-panel-account"
              role="tabpanel"
              style="display: none;"
              tabindex="-1"
            />
            <div
              aria-hidden="false"
              aria-labelledby="rc-tabs-test-tab-mobile"
              class="ant-tabs-tabpane ant-tabs-tabpane-active"
              id="rc-tabs-test-panel-mobile"
              role="tabpanel"
              style=""
              tabindex="0"
            />
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item ant-form-item-with-help ant-form-item-has-error"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
              >
                <span
                  class="ant-input-prefix"
                >
                  <span
                    aria-label="mobile"
                    class="anticon anticon-mobile avx-user-panel-login-prefixIcon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="mobile"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M744 62H280c-35.3 0-64 28.7-64 64v768c0 35.3 28.7 64 64 64h464c35.3 0 64-28.7 64-64V126c0-35.3-28.7-64-64-64zm-8 824H288V134h448v752zM472 784a40 40 0 1080 0 40 40 0 10-80 0z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  class="ant-input ant-input-lg"
                  id="mobile"
                  placeholder="请输入手机号"
                  type="text"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                    role="button"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
          <div
            class="ant-form-item-explain ant-form-item-explain-error"
          >
            <div
              role="alert"
            >
              手机号是必填项！
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                style="display: flex; align-items: center;"
              >
                <span
                  class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
                  style="flex: 1; transition: width .3s; margin-right: 8px;"
                >
                  <span
                    class="ant-input-prefix"
                  >
                    <span
                      aria-label="mail"
                      class="anticon anticon-mail avx-user-panel-login-prefixIcon"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="mail"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0068.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
                        />
                      </svg>
                    </span>
                  </span>
                  <input
                    class="ant-input ant-input-lg"
                    id="captcha"
                    placeholder="请输入验证码"
                    type="text"
                    value=""
                  />
                </span>
                <button
                  ant-click-animating-without-extra-node="true"
                  class="ant-btn ant-btn-lg"
                  style="display: block;"
                  type="button"
                >
                  <span>
                    获取验证码
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 24px;"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
        >
          <span
            class="ant-checkbox ant-checkbox-checked"
          >
            <input
              checked=""
              class="ant-checkbox-input"
              id="autoLogin"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span>
            自动登录
          </span>
        </label>
      </div>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg"
        style="width: 100%;"
        type="button"
      >
        <span>
          登 录
        </span>
      </button>
    </form>
  </div>
</div>
`;

exports[`Login 组件 默认状态 1`] = `
<div>
  <div
    class="avx-user-panel-login-container"
  >
    <form
      autocomplete="off"
      class="ant-form ant-form-vertical"
    >
      <input
        style="display: none;"
        type="text"
      />
      <div
        class="ant-tabs ant-tabs-top ant-tabs-card"
      >
        <div
          class="ant-tabs-nav"
          role="tablist"
        >
          <div
            class="ant-tabs-nav-wrap"
          >
            <div
              class="ant-tabs-nav-list"
              style="transform: translate(0px, 0px);"
            >
              <div
                class="ant-tabs-tab ant-tabs-tab-active"
              >
                <div
                  aria-controls="rc-tabs-test-panel-account"
                  aria-selected="true"
                  class="ant-tabs-tab-btn"
                  id="rc-tabs-test-tab-account"
                  role="tab"
                  tabindex="0"
                >
                  账户密码登录
                </div>
              </div>
              <div
                class="ant-tabs-tab"
              >
                <div
                  aria-controls="rc-tabs-test-panel-mobile"
                  aria-selected="false"
                  class="ant-tabs-tab-btn"
                  id="rc-tabs-test-tab-mobile"
                  role="tab"
                  tabindex="0"
                >
                  手机号登录
                </div>
              </div>
              <div
                class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              />
            </div>
          </div>
          <div
            class="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
          >
            <button
              aria-controls="rc-tabs-test-more-popup"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-hidden="true"
              class="ant-tabs-nav-more"
              id="rc-tabs-test-more"
              style="visibility: hidden; order: 1;"
              tabindex="-1"
              type="button"
            >
              <span
                aria-label="ellipsis"
                class="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          class="ant-tabs-content-holder"
        >
          <div
            class="ant-tabs-content ant-tabs-content-top"
          >
            <div
              aria-hidden="false"
              aria-labelledby="rc-tabs-test-tab-account"
              class="ant-tabs-tabpane ant-tabs-tabpane-active"
              id="rc-tabs-test-panel-account"
              role="tabpanel"
              tabindex="0"
            />
            <div
              aria-hidden="true"
              aria-labelledby="rc-tabs-test-tab-mobile"
              class="ant-tabs-tabpane"
              id="rc-tabs-test-panel-mobile"
              role="tabpanel"
              style="display: none;"
              tabindex="-1"
            />
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg"
              >
                <span
                  class="ant-input-prefix"
                >
                  <span
                    aria-label="user"
                    class="anticon anticon-user avx-user-panel-login-prefixIcon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="user"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  class="ant-input ant-input-lg"
                  id="userName"
                  placeholder="用户名"
                  type="text"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                    role="button"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-row ant-form-item"
      >
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <span
                class="ant-input-affix-wrapper ant-input-affix-wrapper-lg ant-input-password ant-input-password-large"
              >
                <span
                  class="ant-input-prefix"
                >
                  <span
                    aria-label="lock"
                    class="anticon anticon-lock avx-user-panel-login-prefixIcon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="lock"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M832 464h-68V240c0-70.7-57.3-128-128-128H388c-70.7 0-128 57.3-128 128v224h-68c-17.7 0-32 14.3-32 32v384c0 17.7 14.3 32 32 32h640c17.7 0 32-14.3 32-32V496c0-17.7-14.3-32-32-32zM332 240c0-30.9 25.1-56 56-56h248c30.9 0 56 25.1 56 56v224H332V240zm460 600H232V536h560v304zM484 701v53c0 4.4 3.6 8 8 8h40c4.4 0 8-3.6 8-8v-53a48.01 48.01 0 10-56 0z"
                      />
                    </svg>
                  </span>
                </span>
                <input
                  action="click"
                  class="ant-input ant-input-lg"
                  id="password"
                  placeholder="密码"
                  type="password"
                  value=""
                />
                <span
                  class="ant-input-suffix"
                >
                  <span
                    aria-label="eye-invisible"
                    class="anticon anticon-eye-invisible ant-input-password-icon"
                    role="img"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="eye-invisible"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 000-51.5zm-63.57-320.64L836 122.88a8 8 0 00-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 000 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 000 11.31L155.17 889a8 8 0 0011.31 0l712.15-712.12a8 8 0 000-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 00-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 01146.2-106.69L401.31 546.2A112 112 0 01396 512z"
                      />
                      <path
                        d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 00227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 01-112 112z"
                      />
                    </svg>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin-bottom: 24px;"
      >
        <label
          class="ant-checkbox-wrapper ant-checkbox-wrapper-checked"
        >
          <span
            class="ant-checkbox ant-checkbox-checked"
          >
            <input
              checked=""
              class="ant-checkbox-input"
              id="autoLogin"
              type="checkbox"
              value=""
            />
            <span
              class="ant-checkbox-inner"
            />
          </span>
          <span>
            自动登录
          </span>
        </label>
      </div>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg"
        style="width: 100%;"
        type="button"
      >
        <span>
          登 录
        </span>
      </button>
    </form>
  </div>
</div>
`;
